clients/admin-ui/src/features/system/VendorSelector.tsx

```diff
@@ -130,165 +132,194 @@ const VendorSelector = ({
+              handleSelectChange(newValue, actionMeta)
+            }
+            onInputChange={(e) => setSearchParam(e)}
+            inputValue={searchParam}
+            size="sm"
+            onKeyDown={(e) => {
+              if (e.key === "Tab") {
+                handleTabPressed();
+              }
+            }}
+            tabSelectsValue={hasVendorSuggestions}
+            classNamePrefix="custom-select"
+            placeholder="Enter system name..."
+            instanceId="select-name"
+            isDisabled={!isCreate && lockedForGVL}
+            menuPosition="absolute"
+            isSearchable
+            isClearable={!!searchParam}
+            focusBorderColor="primary.600"
+            formatCreateLabel={(inputValue) =>
+              `Create new system "${inputValue}"...`
+            }
+            chakraStyles={{
+              container: (provided) => ({
+                ...provided,
+                flexGrow: 1,
+                backgroundColor: "white",
+              }),
+              option: (provided, state) => ({
+                ...provided,
+                background:
+                  state.isSelected || state.isFocused ? "gray.50" : "unset",
+              }),
+              menu: (provided) => ({
+                ...provided,
+                visibility: hasVendorSuggestions ? "visible" : "hidden",
+              }),
+              dropdownIndicator: (provided) => ({
+                ...provided,
+                display: "none",
+              }),
+              indicatorSeparator: (provided) => ({
+                ...provided,
+                display: "none",
+              }),
+            }}
+            components={{ Option: CustomDictOption }}
+          />
+        </Box>
+        <ErrorMessage
+          isInvalid={isInvalid}
+          message={meta.error}
+          fieldName="name"
         />
-      </HStack>
-      {fieldsSeparated ? (
-        <DictSuggestionTextInput
-          id="name"
-          name="name"
-          dictField={(vendor) => vendor.name ?? (vendor.legal_name || "")}
-          isRequired
-          label="System name"
-          tooltip="Give the system a unique, and relevant name for reporting purposes. e.g. “Email Data Warehouse”"
+        <Text
+          aria-hidden
+          style={{
+            position: "absolute",
+            backgroundColor: "transparent",
+            borderColor: "transparent",
+            marginTop: "31.52px",
+            marginLeft: "13px",
+            pointerEvents: "none",
+            fontSize: "14px",
+            zIndex: 1,
+          }}
+        >
+          <span style={{ color: "transparent" }}>{searchParam}</span>
+          {searchParam && suggestions.length > 0 ? (
+            <span style={{ color: "#824EF2" }}>
+              {suggestions[0].label.substring(searchParam.length)}
+            </span>
+          ) : null}
+        </Text>
+      </VStack>
+    </FormControl>
+  );
+
+  const textInput = (
```

Comment:
it's a little weird for this to be a constant as opposed to a separate React component with props passed in. not a huge deal, but I think

```js
{isTypeahead ? <TypeaheadSelect isInvalid={isInvalid} ... /> : <TextInput isInvalid={isInvalid} ... />}
```

is a little clearer that we are expecting components. if it feels like we are repeating too many of the same props, can also do

```js
const props = { isInvalid: false }
{isTypeahead ? <TypeaheadSelect {...props} /> : <TextInput {...props} />}
```

Reply:
can this component use the `CustomInput` shared one from `inputs.tsx`? or is it too customized?

Reply:
The only real difference with the `CustomInput` is that this one is wrapped in an `InputGroup` with the clear indicator as an `InputRightComponent`-- do you think it's reasonable to add handling for that case into the generic input?

Reply:
I think it could be! maybe if the default input is wrapped in an `InputGroup` (I don't _think_ that actually renders anything but I might be wrong about that), then it could take an optional `InputRightElement`? it's always a balance though—we don't want to add anything too bespoke to the common components, but I think having a right element could be useful for other inputs too so if it doesn't make the code too much more complicated, I think it's worth a shot

Reply:
@allisonking After messing with putting the typeahead into its own component I'm sort of inclined to think that it's not worth it-- both components need access to things from the shared state and from `useFormikContext` and `useField`, so I'd either have to duplicate those hook calls or pass in a whole bunch (like, on the order of 10) of one-off helper methods to the typeahead as props, which I think makes it harder to read.

Reply:
okay, thanks for looking into it!